packages/riipen-ui/src/components/Menu.jsx

```diff
@@ -0,0 +1,168 @@
+import React from "react";
+
+import PropTypes from "prop-types";
+
+import clsx from "clsx";
+
+import MenuList from "./MenuList";
+import MenuItem from "./MenuItem";
+import Popover from "./Popover";
+
+class Menu extends React.Component {
+  static propTypes = {
+    /**
+     * The content of the component.
+     */
+    children: PropTypes.arrayOf(
+      PropTypes.shape({
+        type: PropTypes.oneOf([MenuItem])
+      })
+    ),
+
+    /**
+     * The element the menu is attached too
+     */
+    anchorEl: PropTypes.object,
+
+    /**
+     * The location to attach the content too on the anchor element
+     */
+    anchorPosition: PropTypes.object,
+
+    /**
+     * The location to attach the anchor to on the content element
+     */
+    contentPosition: PropTypes.object,
+
+    /**
+     * The function callback to use when the menu closes
+     */
+    onClose: PropTypes.func,
+
+    /**
+     * The index of the item selected in the list
+     */
+    selectedIndex: PropTypes.number,
+
+    /**
+     * Whether to autofocus the first element on first open
+     */
+    autoFocus: PropTypes.bool,
+
+    /**
+     * The function callback for when the selection is changed
+     */
+    selectChange: PropTypes.func,
+
+    /**
+     * The color of the component. It supports those theme colors that make sense for this component.
+     */
+    color: PropTypes.oneOf(["primary", "secondary"]),
+
+    /**
+     * Array or string of additional CSS classes to use.
+     *
+     * @type {string | Array}
+     */
+    classes: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
+
+    /**
+     * Whether or not the menu should close when an option is chosen with a click event
+     */
+    closeOnClick: PropTypes.bool
+  };
+
+  static defaultProps = {
+    color: "primary",
+    classes: [],
+    closeOnClick: true
+  };
+
+  constructor(props) {
+    super(props);
+    const { selectedIndex = -1 } = props;
+    this.state = {
+      onCloseHandler: this.closeEvent,
+      activeItemIndex: selectedIndex
```

Comment:
I believe you could refactor this component to not maintain this piece of state locally. I could be wrong but if `selectedIndex` is passed in as a prop and you also update it higher in the tree with `selectChanged` you should be able to just use the prop everywhere and bypass state altogether!